### ISSUE TITLE
[Feature] Enable Geo points for device trackers

### ISSUE THREAD

**Issue by r-oleary** [legrego/homeassistant-elasticsearch#106]
I'm totally new to Elastic, but would be cool if our device trackers, which have lat/lon as an attribute, were available to use as geo points in a map in Elastic.

**Comment by legrego** [comment]
Hey @r-oleary thanks for opening this issue. We provide a `hass.geo` field today if we detect lat/lon coordinates, with the intent that you could use it within a map.

Can you check to see if that exists for you, and is populating the way you'd expect?

**Comment by r-oleary** [comment]
I tried to use it and didn't see anything show up on the map, but like I
said I'm new to Elastic so might be doing it wrong.

Also it *appears  *that is only for the location of the hass install
itself? And what I'm looking for is to enable the location data that is in
device trackers.

**Comment by legrego** [comment]
> what I'm looking for is to enable the location data that is in
device trackers.

@r-oleary sorry for the delay. I tested this using my own device trackers, and I was able to use the Maps app to plot the changes in device location over time:

![image](https://user-images.githubusercontent.com/3493255/92330191-0ec8d180-f03b-11ea-85bf-c8c5211358f2.png)

Can you share screenshots of what you're trying to do?

**Comment by WaaZaa666** [comment]
I was just working on trying to do this, but it seems in Elastic 7.11, the geo_point are not working again. 

The template does contain a host.geo.location mapping, but the incoming documents are not mapped as geo_point:
```
        "_source" : {
          "host.geo.location" : {
            "lon" : 4.xxxx,
            "lat" : 51.xxxx
          }
```
Trying to figure out what's causing this. 
v0.4.0-beta3

**Comment by WaaZaa666** [comment]
Now that I found the culprit in my new installation, I'd like to comment this is working very nicely. 
The homeassistant app nicely send the geo information, which the component sends to Elastic. I am able to plot the geo data on the kibana maps visualization. 

So, looks allright to me!

**Comment by legrego** [comment]
@WaaZaa666 I'm glad you geo points working! Did this component do something wrong that you needed to manually correct?

**Comment by WaaZaa666** [comment]
Yeah, firstly, I needed to rename the folder from elastic to elasticsearch to get it working at all. For specific geo_point, I found the index mapping was not loaded. Also, then found alias and ILM policy was not assigned to the index.

To get it working now, I just reindexed the data into a new index where I pre-applied the template manually. After deleting the original index, restarting hass, all seemed to be OK. I can try to reproduce with a test hass if you like.

**Comment by legrego** [comment]
> Yeah, firstly, I needed to rename the folder from elastic to elasticsearch to get it working at all. 

Yeah this is unfortunately expected when migrating from 0.3 to the new 0.4 beta series (https://github.com/legrego/homeassistant-elasticsearch/pull/129).

> For specific geo_point, I found the index mapping was not loaded. Also, then found alias and ILM policy was not assigned to the index.

This part is concerning to me, and likely points to a bug in the component somewhere.

> After deleting the original index, restarting hass, all seemed to be OK. I can try to reproduce with a test hass if you like.

If you have the time to reproduce, that would be greatly appreciated!

**Comment by WaaZaa666** [comment]
It's been a while, but I was able to retest this. 

I deleted all the aliases, index templates and Kibana mappings, removed the component in Homeassistant and re-enabled. Right now the hass.geo.location  & host.geo.location are working just fine.

I'm not sure what happend with my previous installation, might have been some kind of issue while switching from the then 'default' version in HACS to the beta version. 

To me, this item seems solved!
